### PR TITLE
Improve command line interface

### PR DESCRIPTION
**Pull request by jcfr**
### Summary

These changes were implemented following the integration of https://github.com/Slicer/Slicer/pull/7063 contributed by @jhlegarreta. These are intended to streamline the integration of project already using the checker provided by "codespell-project".

* By default, search files in current directory
* Initialize default list of suffixes
* Clarify `--dict` help string, and support `--ignore-words` and `-I` alternative
* Add support for excluding files specifying glob patterns
* Improve help string associated with the `--exclude` argument
* Add regular Python docstrings

### Use in Slicer

```
mkvirtualenv codespell
python -m pip install git+https://github.com/jcfr/SimpleITKSpellChecking@improve-command-line-interface
```

Then apply this patch

```diff
diff --git a/Utilities/Scripts/runCodespell.sh b/Utilities/Scripts/runCodespell.sh
index 6cfe638836..283606868d 100755
--- a/Utilities/Scripts/runCodespell.sh
+++ b/Utilities/Scripts/runCodespell.sh
@@ -33,4 +33,5 @@ if [[ ! -e ${ignore_words_file} ]]; then
   exit 1
 fi
 
-codespell -w -q6 --skip="${skip}" -I ${ignore_words_file}
+codespell --skip="${skip}" -I ${ignore_words_file} \
+  --miss --verbose --prefix "vtkSlicer" --prefix "qSlicer" --prefix "vtkITK" --prefix "qMRML" --prefix "vtkMRML"
```

.. and run the script:

```
/path/to/Slicer/Utilities/Scripts/runCodespell.sh
```

cc: @lassoan @jamesobutler

### CODE REVIEW

**Review by dave3d — APPROVED** [review]
Looks good to me.  Thanks for these improvements.

File: codespell.py [hunk 1]
```diff
@@ -34,33 +49,17 @@ def splitCamelCase(word):
...
-        return 'text/plain'
 
-
-#
-#  For a regular text file, we don't need to parse it for comments.  We
-#  just pass every line to the spell checked
-#
 def load_text_file(filename):
+    """
+    For a regular text file, we don't need to parse it for comments. We
```
Comment (jhlegarreta) [inline codespell.py:60]: Docstrings should start right after the triple quotes.

File: codespell.py [hunk 2]
```diff
@@ -204,13 +213,20 @@ def parse_args():
     parser.add_argument('--vim', '-V', action='store_true', default=False,
                         dest='vim', help='Output results in vim command format')
 
-    parser.add_argument('--dict', '-d', action='append',
+    parser.add_argument('--dict', '-d', '--ignore-words', '-I', action='append',
                         dest='dict',
-                        help='Add a dictionary (multiples allowed)')
+                        help='File that contains words that will be ignored (multiples allowed).'
```
Comment (jhlegarreta) [inline codespell.py:218]: Not sure what "multiples" means here. Can maybe reworded?

File: codespell.py [hunk 3]
```diff
@@ -204,13 +213,20 @@ def parse_args():
...
+                        ' File must contain 1 word per line.')
 
     parser.add_argument('--exclude', '-e', action='append',
                         dest='exclude',
-                        help='Add exclude regex (multiples allowed)')
+                        help='Specify regex for excluding files (multiples allowed)')
```
Comment (jhlegarreta) [inline codespell.py:223]: Same as above.

File: codespell.py [hunk 4]
```diff
@@ -204,13 +213,20 @@ def parse_args():
...
+
+    parser.add_argument('--skip', '-S', action='append',
+                        help='comma-separated list of files to skip. It '
```
Comment (jhlegarreta) [inline codespell.py:226]: Capitalize "comma".

File: codespell.py [hunk 5]
```diff
@@ -34,33 +49,17 @@ def splitCamelCase(word):
...
-                    '.c': 'text/x-c++',
-                    '.hxx': 'text/x-c++',
-                    '.py': 'text/x-python',
-                    '.ruby': 'text/x-ruby',
-                    '.java': 'text/x-java-source',
-                    '.txt': 'text/plain',
-                    '.rst': 'text/plain',
-                    '.md': 'text/plain',
-                  }
+    """Map `filepath`` extension to file type."""
```
Comment (jhlegarreta) [inline codespell.py:53]: Missing backtick at the beginning of "filepath".

**Review by jhlegarreta — COMMENTED** [review]
Sorry for being late. A few comments about docstrings. Thanks for doing this @jcfr.

File: codespell.py (hunk 1 above)
Comment (jamesobutler) [reply codespell.py:60]: Could utilize [`pydocstyle`](https://pypi.org/project/pydocstyle/) to provide CI feedback about doc string styling. Whether the doc string is on the same line as the triple quotes or not depends on the situation that `pydocstyle` can enforce.

**Comment by jcfr** [comment]
@jamesobutler @jhlegarreta 
* https://github.com/SimpleITK/SimpleITKSpellChecking/pull/26

File: codespell.py (hunk 1 above)
Comment (jhlegarreta) [reply codespell.py:60]: @jamesobutler `pydocstyle` says it follows PEP 257 https://peps.python.org/pep-0257/, which does not seem to encourage having blank likes at the beginning of the docstrings. The `numpy` doc style does not, and `google` doc style does not either. At least by default, or across all their examples.

Comment (jamesobutler) [reply codespell.py:60]: `pydocstyle` recommends blank lines with the triple quotes depending on the situation. It recommends the following:
```python
def myfunction(self):
    """
    Summarize the function in the first line.

    Add more details about the the purpose and reasoning to provide more context in additional lines for the doc string.
    """
```
```python
def myfunction(self):
    """Summarize the function in the first line."""
```

Comment (jhlegarreta) [reply codespell.py:60]: Then that is `pydocstyle` choice, not a PEP 257 recommendation? Anyways, using a consistent style is better.

Comment (jcfr) [reply codespell.py:60]: Will setup GitHub action with black and blacken-docs.

It the meantime, these concerts have been addressed in the follow up pull request.
